### PR TITLE
fix: check directory for secret keys

### PR DESCRIPTION
**Pull request by polaroi8d**
Fix contains the followings:
- We have to check the `EISDIR` error and added the missing error check from `os.ReadFile()`
- Rename `location` to `path`

### CODE REVIEW

File: agent/internal/config/secret.go [hunk 1]
```diff
@@ -28,37 +29,43 @@ func (field *ConfigFromFile) SetValue(location string) error {
 	return nil
 }
 
-func checkGenerateKeys(location string) (string, error) {
-	log.Printf("Checking key file: %v\n", location)
-	file, err := os.ReadFile(location) //#nosec G304 -- location comes from an env
+func checkGenerateKeys(path string) (string, error) {
+	log.Printf("Checking key file: %v\n", path)
+	readFile, err := os.ReadFile(path) //#nosec G304 -- path comes from an env
```
Comment (nandor-magyar) [inline agent/internal/config/secret.go:34]: minor: readFile sounds like a function, fileContent or similar would be a bit better.

File: agent/internal/config/secret.go [hunk 2]
```diff
@@ -8,18 +8,19 @@ import (
...
+	"syscall"
 
 	"github.com/ProtonMail/gopenpgp/v2/crypto"
 )
 
 type ConfigFromFile string
 
-func (field *ConfigFromFile) SetValue(location string) error {
-	if location == "" {
+func (field *ConfigFromFile) SetValue(path string) error {
```
Comment (nandor-magyar) [inline agent/internal/config/secret.go:18]: Using `path` as variable name would conflict/shadow the `path` package from stdlib, it is better be avoided, it is painful when it happens.

File: agent/internal/config/secret.go [hunk 3]
```diff
@@ -28,37 +29,43 @@ func (field *ConfigFromFile) SetValue(location string) error {
...
 }
 
-func checkGenerateKeys(location string) (string, error) {
-	log.Printf("Checking key file: %v\n", location)
-	file, err := os.ReadFile(location) //#nosec G304 -- location comes from an env
+func checkGenerateKeys(path string) (string, error) {
+	log.Printf("Checking key file: %v\n", path)
+	readFile, err := os.ReadFile(path) //#nosec G304 -- path comes from an env
+
+	if errors.Is(err, syscall.EISDIR) {
```
Comment (robot9706) [inline agent/internal/config/secret.go:36]: Chaining the if-s (if-else if-else) would look better than the current if if-else. Opinions?

**Review by robot9706 — CHANGES_REQUESTED** [review]

**Review by nandor-magyar — APPROVED** [review]

File: agent/internal/config/secret.go (hunk 3 above)
Comment (polaroi8d) [reply agent/internal/config/secret.go:36]: Talked in person.

**Review by robot9706 — APPROVED** [review]
LGTM 😎

File: agent/internal/config/secret.go (hunk 2 above)
Comment (polaroi8d) [reply agent/internal/config/secret.go]: Fixed.

File: agent/internal/config/secret.go (hunk 1 above)
Comment (polaroi8d) [reply agent/internal/config/secret.go]: Fixed.

**Pull request merged**